### ISSUE TITLE
init.c compilation warning on CRAN

### ISSUE THREAD

**Issue by bertcarnell** [bertcarnell/lhs#52]
clang-15 -I"/home/hornik/tmp/R.check/r-devel-clang/Work/build/include" -DNDEBUG -DRCOMPILE -I'/home/hornik/tmp/R.check/r-devel-clang/Work/build/Packages/Rcpp/include' -I/usr/local/include -D_FORTIFY_SOURCE=2   -fpic  -g -O3 -Wall -pedantic -c init.c -o init.o
init.c:14:33: warning: a function declaration without a prototype is deprecated in all versions of C [-Wstrict-prototypes]
extern SEXP get_library_versions();
                                ^
                                 void
1 warning generated.

**Comment by bertcarnell** [comment]
This is what I did to try to fix this issue.

1. fork rhub/rhub-linux-builders  (https://github.com/bertcarnell/rhub-linux-builders)
2. build and start the container 

```
docker build -t rhub/fedora fedora
docker build -t rhub/fedora-gcc fedora-gcc
docker build -t rhub/fedora-gcc-devel fedora-gcc-devel
docker run -ti --rm -v /mnt/c/repositories:/home/docker -w /home/docker -u 0 --name build-lhs rhub/fedora-gcc-devel
```

3. inside the container

```
dnf install -y gmp-devel
/opt/R-devel/bin/R
install.packages(c("DoE.base", "knitr", "rmarkdown", "testthat", "assertthat", "Rcpp"), repos = "https://cran.case.edu")
q()
```

4. replicate the error

``` 
gcc -I"/opt/R-devel/lib64/R/include" -DNDEBUG -DRCOMPILE -I'/opt/R-devel/lib64/R/library/Rcpp/include' -I/usr/local/include -fpic -g -O2 -Wall -pedantic -mtune=native -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong -fstack-clash-protection -fcf-protection -Werror=implicit-function-declaration -Wstrict-prototypes -fpic  -g -O2 -c init.c -o init.o
```

5. make the fix by declairing `get_library_versions(void)`
6. verify the fix made the warning resolve
7. also change the function in the CPP file and check it compiles

```
g++ -std=gnu++11 -I"/opt/R-devel/lib64/R/include" -DNDEBUG -DRCOMPILE -I'/opt/R-devel/lib64/R/library/Rcpp/include' -I/usr/local/include    -fpic  -g -O2  -Wall -pedantic -mtune=native -Wno-ignored-attributes -Wno-parentheses -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong -fstack-clash-protection -fcf-protection -c lhs_r.cpp -o lhs_r.o
```

8.  Check the package again

```
/opt/R-devel/bin/R CMD build lhs
/opt/R-devel/bin/R CMD check lhs_1.1.6.tar.gz
```

**Comment by bertcarnell** [comment]
Fixed with 766782b114602bb15bcad42811a84e9178f92e99
